kratos/tests/cpp_tests/sources/test_model_part.cpp

```diff
@@ -580,4 +583,74 @@ namespace Kratos::Testing {
             model_part.RemoveSubModelPart("Inlet1.sub_inlet.sub_sub_inlet.test"),
             "Error: There is no sub model part with name \"sub_sub_inlet\" in model part \"Main.Inlet1.sub_inlet\"");
     }
+
+    KRATOS_TEST_CASE_IN_SUITE(ModelPartSubRangeAddition, KratosCoreFastSuite)
+    {
+        Model model;
+        auto& r_model_part = model.CreateModelPart("test");
+        Properties::Pointer p_elem_prop = r_model_part.CreateNewProperties(0);
+
+        for (IndexType i = 0; i < 16; ++i) {
+            r_model_part.CreateNewNode(i + 1, 0.0, 0.0, 0.0);
+        }
+
+        KRATOS_EXPECT_EQ(r_model_part.Nodes().size(), 16);
+        KRATOS_EXPECT_EQ(r_model_part.Nodes().capacity(), 16);
```

Comment:
No wonder this fails in the CI. Memory management is up to the standard lib's implementation. The only guaranteed way of enforcing a specific capacity is with `shrink_to_fit` and you haven't called that anywhere.

I'd just remove capacity tests altogether.

Reply:
I have changed to check for the change in capacity and used shrink_to_fit to align memory allocations in different compilers.